src/Sync.ts

```diff
@@ -0,0 +1,216 @@
+import { ComponentEvent } from "@egjs/component";
+import Flicking, { EVENTS, Panel, Plugin } from "@egjs/flicking";
+
+import { SYNC } from "./const";
+import { addClass, removeClass } from "./utils";
+
+interface SyncOptions {
+  type: "camera" | "index";
+  synchronizedFlickingOptions: SychronizableFlickingOptions[];
+}
+
+interface SychronizableFlickingOptions {
+  flicking: Flicking;
+  isClickable?: boolean;
+  isSlidable?: boolean;
+  activeClass?: string;
+}
+
+/**
+ * Plugin for synchronizing multiple flickings 
+ * @ko 다양한 형태로 Flicking들이 같이 움직이게 할 수 있습니다.
+ * @memberof Flicking.Plugins
+ */
+class Sync implements Plugin {
+  /* Internal Values */
+  private _flicking: Flicking | null = null;
+
+  /* Options */
+  private _type: SyncOptions["type"];
+  private _synchronizedFlickingOptions: SyncOptions["synchronizedFlickingOptions"];
+
+  public get type() { return this._type; }
+  public get synchronizedFlickingOptions() { return this._synchronizedFlickingOptions; }
+
+  public set type(val: SyncOptions["type"]) {
+    this.preventEvent(() => {
+        this._type = val;
+    });
+  }
+
+  public set synchronizedFlickingOptions(val: SyncOptions["synchronizedFlickingOptions"]) {
+    this.preventEvent(() => {
+      this._synchronizedFlickingOptions = val;
+    });
+  }
+
+  private preventEvent(fn: () => void) {
+    this._removeEvents(this._synchronizedFlickingOptions);
+    fn();
+    this._addEvents(this._synchronizedFlickingOptions);
+  }
+
+  public constructor({
+    type = SYNC.TYPE.CAMERA,
+    synchronizedFlickingOptions = [],
+  }: Partial<SyncOptions> = {}) {
+    this._type = type;
+    this._synchronizedFlickingOptions = synchronizedFlickingOptions;
+  }
+
+  public init(flicking: Flicking): void {
+    if (this._flicking) {
+      this.destroy();
+    }
+
+    this._flicking = flicking;
+
+    this._addEvents(this._synchronizedFlickingOptions);
+
+    this._synchronizedFlickingOptions.forEach(synchronizedFlickingOption => {
+      this._updateClass(synchronizedFlickingOption, 0);
+    });
+  }
+
+  public destroy(): void {
+    const flicking = this._flicking;
+
+    if (!flicking) {
+      return;
+    }
+
+    this._removeEvents(this._synchronizedFlickingOptions);
+
+    this._flicking = null;
+  }
+
+  public update(): void {
+    this._synchronizedFlickingOptions.forEach(synchronizedFlickingOption => {
+      this._updateClass(synchronizedFlickingOption, synchronizedFlickingOption.flicking.camera.position);
+    });
+  }
+
+  private _addEvents = (synchronizedFlickingOptions: SychronizableFlickingOptions[]): void => {
+    synchronizedFlickingOptions.forEach((synchronizedFlickingOption) => {
+      if (this._type === "camera") {
+        synchronizedFlickingOption.flicking.on(EVENTS.MOVE, this._onMove);
```

Comment:
There is a lot of duplicate property use.


use this:
```js
({ flicking, isSlidable, isClickable }) => {} 
```

Reply:
All forEach using synchronizedFlickingOptions has been refactored.